### ISSUE TITLE
Strategies for installing Grafana separately and loading dashboards

### ISSUE THREAD

**Issue by alpeb** [linkerd/linkerd2#7358]
Ref #6045

We'd like to stop providing a docker image based on Grafana, given their license changes.
We could instruct users to install Grafana directly through their Helm chart:
```
helm repo add grafana https://grafana.github.io/helm-charts
helm install grafana -n grafana --create-namespace grafana/grafana -f https://github.com/linkerd/linkerd2/blob/main/grafana/values.yaml
```

where `values.yaml` would be a new file with the default config needed to hook that Grafana instance with Linkerd (pointing to viz' Prometheus and discoverable by `web`).

The challenge is how to load our default dashboards. I can think of these options:

### 1) Upload all the charts to https://grafana.com/grafana/dashboards/

We'd only have to add this to values.yaml to point to those charts, which would be fetched automatically:
```yaml
dashboards:
  default:
    # all these charts are hosted at https://grafana.com/grafana/dashboards/{id}
    top-line:
      gnetId: 14276
      revision: 1
      datasource: prometheus
    deployment:
      gnetId: 14264
      revision: 1
      datasource: prometheus
    pod:
      gnetId: 14269
      revision: 1
      datasource: prometheus
    ...
```

We could still keep the dashboard definitions versioned in the linkerd2 repo under `/grafana/dashboards`, warning folks that those files would need to remain in sync with what gets uploaded to grafana.com.

### 2) Include the dashboard definitions inside values.yaml

The downside here is that it'd end up being a huge file, and we wouldn't be able to track changes in git separately for each dashboard.

### 3) Separate dashboard definitions into ConfigMaps

Create a new Helm chart `linkerd-grafana-dashboards` with a ConfigMap for each dashboard. The values.yaml mentioned above would have the settings necessary to wire thes CMs. Possibly a new `linkerd viz grafana-dashboards` command could be created to render all those CMs as well.

### 4) `linkerd-grafana` extension

As suggested originally in #6045, we could still provide our own Grafana image loaded with all the dashboards, based on an up-to-date upstream Grafana image, as long as provide it through a separate linkerd extension

Thoughts?

**Comment by olix0r** [comment]
All things equal, I'm a fan of the first option (publishing the dashboards to grafana's service). Otherwise, we'd probably be best going with option 4 (splitting this all out into another extension).

**Comment by jack1902** [comment]
@alpeb to throw a 5th contender into the ring. Would it be possible to add https://github.com/grafana-operator/grafana-operator templates with a boolean wrapper (off by default) which point to configMapRefs which are created by point 3.

This means that you still get all the benefits of 3 for git tracking of individual dashboards, whilst making it possible to add the dashboards to an existing grafana deployment that is managed by the grafana-operator without needing to wire them into whichever deployment of grafana people are using

**Comment by alpeb** [comment]
@jack1902 You mean our chart would produce `GrafanaDashboard` CRs instead of ConfigMaps, if enabled by some new flag? Or even better, do you know if there's a way for the grafana operator to consume ConfigMaps instead of `GrafanaDashboard` instances?

**Comment by alpeb** [comment]
Fixed by #7567
